api/python/cellxgene_census/src/cellxgene_census/experimental/pp/_highly_variable_genes.py

```diff
@@ -0,0 +1,338 @@
+        x = np.log10(u[not_const])
+        model = skmisc.loess.loess(x, y, span=span, degree=2)
+        # Can we resolve low entropy loess errors by adding jitter and retrying?
+        # See: https://github.com/has2k1/scikit-misc/issues/9
+        model.fit()
+        estimated_variances[not_const] = model.outputs.fitted_values
+        reg_std[batch] = np.sqrt(10**estimated_variances)
+
+        vmax = np.sqrt(N)
+        clip_val[batch] = reg_std[batch] * vmax + u
+
+    del estimated_variances
+
+    # Read counts again, clip and save sum of counts and sum of counts squared
+    with futures.ThreadPoolExecutor(max_workers=max_workers) as pool:
+        acc = CountsAccumulator(n_batches, query.n_vars, clip_val)
+        for arrow_tbl in EagerIterator(query.X(layer).tables(), pool=pool):
+            data = arrow_tbl["soma_data"].to_numpy()
+            if batch_indexer:
+                _batch_take_at_future = pool.submit(batch_indexer, arrow_tbl["soma_dim_0"])
+                var_dim = var_indexer.by_var(arrow_tbl["soma_dim_1"])
+                _batch_vec = batch_codes[_batch_take_at_future.result()]
+                acc.update_by_batch(_batch_vec, var_dim, data)
+            else:
+                var_dim = var_indexer.by_var(arrow_tbl["soma_dim_1"])
+                acc.update_single_batch(var_dim, data)
+
+        counts_sum, squared_counts_sum = acc.finalize()
+        norm_gene_vars = (1 / ((n_samples - 1) * np.square(reg_std.T))).T * (
+            (n_samples * np.square(batches_u.T)).T + squared_counts_sum - 2 * counts_sum * batches_u
+        )
+        del acc, counts_sum, squared_counts_sum
+
+    ranked_norm_gene_vars = np.argsort(np.argsort(-norm_gene_vars, axis=1), axis=1)
+    ranked_norm_gene_vars = ranked_norm_gene_vars.astype(np.float32)
+    num_batches_high_var = np.sum((ranked_norm_gene_vars < n_top_genes).astype(int), axis=0)
+    ranked_norm_gene_vars[ranked_norm_gene_vars >= n_top_genes] = np.nan
+    ma_ranked = np.ma.masked_invalid(ranked_norm_gene_vars)  # type: ignore[no-untyped-call]
+    median_ranked = np.ma.median(ma_ranked, axis=0).filled(np.nan)  # type: ignore[no-untyped-call]
+
+    var_df = var_df.assign(
+        highly_variable_nbatches=pd.Series(num_batches_high_var, index=var_df.index),
+        highly_variable_rank=pd.Series(median_ranked, index=var_df.index),
+        variances_norm=pd.Series(np.mean(norm_gene_vars, axis=0), index=var_df.index),
+    )
+
+    sorted_index = (
+        var_df[["highly_variable_rank", "highly_variable_nbatches"]]
+        .sort_values(
+            ["highly_variable_rank", "highly_variable_nbatches"],
+            ascending=[True, False],
+            na_position="last",
+        )
+        .index
+    )
+    var_df["highly_variable"] = False
+    var_df.loc[sorted_index[: int(n_top_genes)], "highly_variable"] = True
+    if batch_key is None:
+        var_df = var_df.drop(columns=["highly_variable_nbatches"])
+    return var_df
+
+
+def highly_variable_genes(
```

Comment:
nit: better name to differentiate it from the convenience wrapper method, below? `compute_highly_variable_genes`? I don't have any great naming suggestions, so feel free to leave as is.